extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -115,6 +114,52 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 			telemetryStep = CreateAzureFunctionStep.launchFromCommandPalette;
 
 			// prompt user for connection profile to get connection info
+			while (true) {
+				connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+				if (!connectionInfo) {
+					// User cancelled
+					return;
+				}
+				telemetryStep = 'getConnectionInfo';
+				let connectionURI: string = '';
+				try {
+					await vscode.window.withProgress(
+						{
+							location: vscode.ProgressLocation.Notification,
+							title: constants.connectionProgressTitle,
+							cancellable: false
+						}, async (_progress, _token) => {
+							// list databases based on connection profile selected
+							connectionURI = await vscodeMssqlApi.connect(connectionInfo!);
+						}
+					);
+				} catch (e) {
+					// mssql connection error will be shown to the user
+					// we will then prompt user to choose a connection profile again
+					continue;
+				}
+				// list databases based on connection profile selected
+				let listDatabases = await vscodeMssqlApi.listDatabases(connectionURI);
+				const selectedDatabase = (await vscode.window.showQuickPick(listDatabases, {
+					canPickMany: false,
+					title: constants.selectDatabase,
+					ignoreFocusOut: true
+				}));
+
+				if (!selectedDatabase) {
+					// User cancelled
+					continue;
```

Comment:
You're still doing the continue  here - is that intentional? If so please add a comment explaining why we want to continue in this case.

Reply:
Yeah, I would imagine if the user cancels the database prompt we can have them show the connection profile prompt once more for them to either choose another connection where their database was in instead for example.